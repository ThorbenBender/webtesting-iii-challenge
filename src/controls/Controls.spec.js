// Test away!
import React from 'react';
import * as rt from 'react-testing-library';
import Control from './Controls';

function setUpWrap(props = {}) {
	rt.cleanup();
	return rt.render(<Control {...props} />);
}

afterEach(rt.cleanup);

describe('Control', () => {
	let wrap;
	beforeEach(() => {
		wrap = setUpWrap();
	});
	it('displays button', () => {
		const button = wrap.queryByTestId(/Button/i);
		expect(button);
		expect(wrap.asFragment()).toMatchSnapshot();
	});
	it('Control', () => {
		const wrapper = rt.render(<Control />);
	});
});
