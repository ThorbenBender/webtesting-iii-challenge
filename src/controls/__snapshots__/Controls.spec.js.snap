// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control displays button 1`] = `
<DocumentFragment>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      data-testid="Button"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="Button"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
